### DISCUSSION TITLE
Snap lens tool - Fixing infinite lens load

### DISCUSSION THREAD

**Discussion by JmckT** [ptrumpis/snap-camera-server#86] [Q&A]
Sorry! I am a bit dense when it comes to the pc things. I cant figure out how to use the lens tool to fix the infinite load on the lens I was wanting to use. Help please?

Thanks in advance!

**Comment by ptrumpis** [comment]
The reason for infinite loading will be in your log file
- Windows:
  `%USERPROFILE%\AppData\Local\Snap\Snap Camera\Log.txt `
- Mac OS:
  `/Users/{YOUR_USER_NAME}/Library/Preferences/Snap/Snap Camera/Log.txt`

**Reply by JmckT** [reply]
So I found the log. What am I looking for exactly? Thanks again.

**Reply by ptrumpis** [reply]
After selecting the infinite loading lens the error message will appear inside the log file along with some other debugging output.
This can be:
- "Failed to get remote asset"
- "Lens not loaded exception: Unknown blend mode"
- "Checksum doesn't match"

or similiar

**Reply by JmckT** [reply]
Update and more info-
I’m looking to use the Stunning look lens. The requirements.txt tool ran perfectly. When trying to run the .py snap lens tool- I ran into problems. 

I definitely have python installed and on my path, but it keeps throwing “DLL load failed while importing QtWidgets: The specified module could not be found. “

**Reply by ptrumpis** [reply]
You may have to restart (close/re-open) the Terminal window after installing requirements.txt
This will probably fix the error.

**Reply by BlacKarmaz** [reply]
I have the same issue but i'm not really a computer guy is there a visual tutorial for that, or can you guide me step by step on how to insert that text? please and thank you.

**Comment by Orpheusll** [comment]
I'm not quite sure on how to use the Snap Lens tool exactly either. Is there any guide available here or can you write it down as bullet points or steps? Highly appreciate it

**Reply by ptrumpis** [reply]
Can you clarify what you are trying to accomplish and which part you are struggling with?
Did you manage to run the tool?

**Reply by Orpheusll** [reply]
i've got the tool running now finally!  I am not sure how to get a lens to work after it though.  (Apparently my python versions were clashing with each other)

can you tell me what I would have to do for example to get this one to work: https://www.snapchat.com/lens/22ca26aaa0b841af8c1e30968fab7c0f?type=SNAPCODE&metadata=01&

Through snap lens tool? and does it even work after it or am I just trying a wrong lens
